packages/stacking/src/index.ts

```diff
@@ -0,0 +1,410 @@
+// import axios from 'axios';
+// import { BigNumber } from 'bignumber.js';
+import {
+  makeContractCall,
+  bufferCV,
+  uintCV,
+  tupleCV,
+  ClarityType,
+  broadcastTransaction,
+  standardPrincipalCV,
+  ContractCallOptions,
+  UIntCV,
+  BufferCV,
+  ContractCallPayload,
+  StacksTransaction,
+  callReadOnlyFunction,
+  cvToString,
+  ClarityValue,
+  ResponseErrorCV,
+  SomeCV,
+  TupleCV
+} from '@stacks/transactions';
+import {
+  StacksNetwork,
+  StacksMainnet
+} from '@stacks/network';
+import BN from 'bn.js';
+import { StackingErrors } from './constants';
+import { fetchPrivate } from '@stacks/common';
+import { convertBTCAddress } from './utils';
+
+export interface PoxInfo {
+  contract_id: string;
+  first_burnchain_block_height: number;
+  min_amount_ustx: string;
+  registration_window_length: 250;
+  rejection_fraction: number;
+  reward_cycle_id: number;
+  reward_cycle_length: number;
+  rejection_votes_left_required: number;
+}
+
+export interface StackerInfo {
+  amountMicroStx: string;
+  firstRewardCycle: number;
+  lockPeriod: number;
+  poxAddress: {
+    version: Buffer;
+    hashbytes: Buffer;
+  };
+  btcAddress: string;
+}
+
+export interface BlockTimeInfo {
+  mainnet: {
+    target_block_time: number;
+  },
+  testnet: {
+    target_block_time: number;
+  }
+}
+
+export interface CoreInfo {
+  burn_block_height: number;
+  stable_pox_consensus : string;
+}
+
+export interface BalanceInfo {
+  balance: string;
+  nonce: number;
+}
+
+export interface StackingEligibility {
+  eligible: boolean,
+  reason?: string
+}
+
+/**
+ * Lock stx check options
+ *
+ * @param  {String} poxAddress - the reward Bitcoin address
+ * @param  {number} cycles - number of cycles to lock
+ */
+export interface CanLockStxOptions {
+  poxAddress: string, 
+  cycles: number
+}
+
+/**
+ * Lock stx options
+ *
+ * @param  {String} key - private key to sign transaction
+ * @param  {String} poxAddress - the reward Bitcoin address
+ * @param  {number} cycles - number of cycles to lock
+ * @param  {BigNum} amountMicroStx - number of microstacks to lock
+ * @param  {number} burnBlockHeight -the burnchain block height to begin lock
+ */
+export interface LockStxOptions {
+  key: string;
```

Comment:
could we be more descriptive here? something like `stackerPrivateKey`?

Reply:
Changed to `privateKey`